state/cloudcredentials.go

```diff
@@ -115,7 +129,21 @@ func (st *State) RemoveCloudCredential(tag names.CloudCredentialTag) error {
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		return removeCloudCredentialOps(tag), nil
+		ops := removeCloudCredentialOps(tag)
+
+		cloudCredKey := cloudCredentialGlobalKey(tag)
+		if !force {
+			refcounts, closer := st.db().GetCollection(globalrefcountsC)
+			defer closer()
+			refOp, err := nsRefcounts.RemoveOp(refcounts, cloudCredKey, 0)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			ops = append(ops, refOp)
+		} else {
+			ops = append(ops, nsRefcounts.JustRemoveOp(globalrefcountsC, cloudCredKey, -1))
```

Comment:
I'd like to address this in a followup, where I'll need to add a state method for looking up models by cloud credential for a facade method.